### PR TITLE
Bump k8s.io/kube-openapi to remove archived dependency

### LINKED ISSUE #4962: PuerkitoBio/urlesc  is archived
follow up of https://github.com/kubernetes/kubernetes/pull/114912

https://github.com/kubernetes-sigs/kustomize/blob/9bc75c16d922f0ee78f6e7351e41d39ae6b8ca69/api/go.mod#L20

github.com/PuerkitoBio/urlesc

depends on it indirectlly
-         "k8s.io/kube-openapi",
-         "sigs.k8s.io/kustomize/api",
-         "sigs.k8s.io/kustomize/kustomize/v4",
-         "sigs.k8s.io/kustomize/kyaml"

### PR DESCRIPTION
**Pull request by olivierlemasle**
Bump `k8s.io/kube-openapi` to remove archived dependency `github.com/PuerkitoBio/urlesc`

Fixes #4962

### CODE REVIEW

File: cmd/gorepomod/go.mod [hunk 1]
```diff
@@ -2,6 +2,4 @@ module sigs.k8s.io/kustomize/cmd/gorepomod
 
 go 1.19
 
-require golang.org/x/mod v0.4.2
-
-require golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
+require golang.org/x/mod v0.7.0
```
Comment (KnVerey) [inline cmd/gorepomod/go.mod:5]: Is this bump related somehow?

File: api/go.mod [hunk 2]
```diff
@@ -8,32 +8,29 @@ require (
 	github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
 	github.com/imdario/mergo v0.3.6
 	github.com/pkg/errors v0.9.1
-	github.com/stretchr/testify v1.8.0
+	github.com/stretchr/testify v1.8.1
```
Comment (KnVerey) [inline api/go.mod:11]: Is this bump related somehow? Usually we want each dep in its own PR, but in this case I don't mind too much because testify is a tests-only dep.

**Review by KnVerey — COMMENTED** [review]
For the record the diff is https://github.com/kubernetes/kube-openapi/compare/b28bf28186616149243529898013e7dfe00d3b3c...3758b55a6596. Since whatever k/k is using for this dependency is canonical for us too, and our tests pass, I'm not concerned.

**Comment by olivierlemasle** [comment]
Yes @KnVerey, the only command I passed was:

```
hack/for-each-module.sh "go list -m -u k8s.io/kube-openapi && go get k8s.io/kube-openapi@3758b55a65962062abe14021c50eade1475f67e3 && go mod tidy"
```

The other modules that were upgraded in the process (`github.com/stretchr/testify`, `golang.org/x/mod`) were upgraded automatically as dependencies of `k8s.io/kube-openapi`.

**Comment by olivierlemasle** [comment]
@KnVerey I just rebased on master, because of merge conflicts.
